File: cs205/DAA_1&2/py_programs/BSB.py
import datetime
def bubblesort(arr):
    n = len(A)
    for i in range(n - 1, 0, -1):
        swapped = False
        for j in range(i):
            if A[j]>A[j+1]:
                swapped = True
                A[j] = A[j+1] + A[j]
                A[j+1] = A[j] - A[j+1]
                A[j] = A[j] - A[j+1]
        if not swapped:
            return

print("Best Case: Array is sorted")
for i in range(1,11):
    print("File",i)
    file = open("..\\..\\Files\\File "+str(i)+".txt","r")
    A = list(map(int,file.read().split()))
    file.close()
    A.sort()
    start = datetime.datetime.now()
    bubblesort(A)
    end = datetime.datetime.now()
    # for i in A:
    #     print(i,end = ' ')
    Ttime = float((end-start).total_seconds()*1000)
    print("Best case time:",Ttime,"milliseconds")

File: cs205/DAA_1&2/cpp_programs/BSB.cpp
#include<bits/stdc++.h>
using namespace std;

void bubblesort(vector<int>& A){
    long long int n = A.size();
    for(int i=n-1; i>0; i--){
        bool swapped = false;
        for(int j=0;j<i;j++){
            if(A[j]>A[j+1]){
                swapped = true;
                A[j] = A[j+1] + A[j];
                A[j+1] = A[j] - A[j+1];
                A[j] = A[j] - A[j+1];
            }
        }
        if(!swapped) return;
    }
}

int main(){
    cout<<"Best Case: sorted array"<<endl;
    for(int i=0; i<10; i++){
        ifstream file("..\\..\\Files\\File "+to_string(i+1)+".txt");
        vector<int> numbers;
        int num;
        while(file>>num){
            numbers.push_back(num);
        }
        sort(numbers.begin(),numbers.end());
        cout<<"File "<<i+1<<endl;
        auto start = chrono::high_resolution_clock::now();
        bubblesort(numbers);
        auto end = chrono::high_resolution_clock::now();
        chrono::duration<double,milli> time = end-start;
        cout<<"Best case time: "<<time.count()<<"ms"<<endl;
    }
}